services/manifest-repo-export-service/pkg/repository/transformer_test.go

```diff
@@ -3021,3 +3021,278 @@ func TestUndeployLogic(t *testing.T) {
+					TransformerMetadata: TransformerMetadata{
+						AuthorName:  authorName,
+						AuthorEmail: authorEmail,
+					},
+					TransformerEslVersion: 1,
+				},
+				&DeleteEnvironment{
+					Environment: envAcceptance,
+					TransformerMetadata: TransformerMetadata{
+						AuthorName:  authorName,
+						AuthorEmail: authorEmail,
+					},
+					TransformerEslVersion: 2,
+				},
+			},
+			expectedMissing: []*FilenameAndData{
+				{
+					path:     "/environments/acceptance",
+					fileData: []byte(authorEmail),
+				},
+				{
+					path:     "/argocd/v1alpha1/acceptance.yaml",
+					fileData: []byte(authorEmail),
+				},
+			},
+			expectedMessage: "delete environment \"acceptance\"",
+		},
+		{
+			Name: "create two environments and delete one of them",
+			Transformers: []Transformer{
+				&CreateEnvironment{
+					Environment: envAcceptance,
+					Config:      envAcceptanceConfig,
+					TransformerMetadata: TransformerMetadata{
+						AuthorName:  authorName,
+						AuthorEmail: authorEmail,
+					},
+					TransformerEslVersion: 1,
+				},
+				&CreateEnvironment{
+					Environment: envAcceptance2,
+					Config:      envAcceptance2Config,
+					TransformerMetadata: TransformerMetadata{
+						AuthorName:  authorName,
+						AuthorEmail: authorEmail,
+					},
+					TransformerEslVersion: 2,
+				},
+				&DeleteEnvironment{
+					Environment: envAcceptance,
+					TransformerMetadata: TransformerMetadata{
+						AuthorName:  authorName,
+						AuthorEmail: authorEmail,
+					},
+					TransformerEslVersion: 3,
+				},
+			},
+			expectedMissing: []*FilenameAndData{
+				{
+					path:     "/environments/acceptance",
+					fileData: []byte(authorEmail),
+				},
+				{
+					path:     "/argocd/v1alpha1/acceptance.yaml",
+					fileData: []byte(authorEmail),
+				},
+			},
+			expectedData: []*FilenameAndData{
+				{
+					path: "/environments/acceptance2/config.json",
+					fileData: []byte(`{
+  "upstream": {
+    "environment": "acceptance2",
+    "latest": true
+  },
+  "argocd": {
+    "destination": {
+      "name": "",
+      "server": ""
+    }
+  }
+}
+`),
+				},
+				{
+					path: "/argocd/v1alpha1/acceptance2.yaml",
+					fileData: []byte(`apiVersion: argoproj.io/v1alpha1
+kind: AppProject
+metadata:
+  name: acceptance2
+spec:
+  description: acceptance2
+  destinations:
+  - {}
+  sourceRepos:
+  - '*'
+`),
+				},
+			},
+			expectedMessage: "delete environment \"acceptance\"",
+		},
+		{
+			Name: "delete an environment that does not exist",
+			Transformers: []Transformer{
+				&DeleteEnvironment{
+					Environment: envAcceptance,
+					TransformerMetadata: TransformerMetadata{
+						AuthorName:  authorName,
+						AuthorEmail: authorEmail,
+					},
+					TransformerEslVersion: 1,
+				},
+			},
+			expectedError: errMatcher{msg: `error within transaction: first apply failed, aborting: error at index 0 of transformer batch: error deleting the environment directory "environments/acceptance": file does not exist`},
```

Comment:
we should differentiate 2 cases:
1) Can't delete because file does not exist. If that's the case, nothing is really wrong, so we should continue normally.
2) Can't delete because any other issue. Then we need to return an error.

Please make 2 test cases out of this.